src/archivist/recorder/repositories/git/index.test.js

```diff
@@ -712,4 +712,188 @@ describe('GitRepository', () => {
       expect(fetchDates).to.deep.equal([ FETCH_DATE_EARLIER, FETCH_DATE, FETCH_DATE_LATER ]);
     });
   });
+
+  context('backwards compatibility with deprecated commit messages', () => {
+    const expectedIds = [];
+    const expectedDates = [];
+
+    let subject;
+
+    const commits = {
+      deprecatedFirstRecord: {
+        path: 'service/terms-deprecated.md',
+        content: 'content',
+        message: 'Start tracking Service Terms',
+        date: new Date('2023-02-28T01:00:00.000Z'),
+      },
+      deprecatedRefilter: {
+        path: 'service/terms-deprecated.md',
+        content: 'content refiltered',
+        message: 'Refilter Service Terms',
+        date: new Date('2023-02-28T02:00:00.000Z'),
+      },
+      deprecatedUpdate: {
+        path: 'service/terms-deprecated.md',
+        content: 'content updated',
+        message: 'Update Service Terms',
+        date: new Date('2023-02-28T03:00:00.000Z'),
+      },
+      currentFirstRecord: {
+        path: 'service/terms-current.md',
+        content: 'content',
+        message: 'Initial record of Service Terms',
+        date: new Date('2023-02-28T04:00:00.000Z'),
+      },
+      currentExtractOnly: {
+        path: 'service/terms-current.md',
+        content: 'content extract only',
+        message: 'Apply technical or declaration upgrade on Service Terms',
+        date: new Date('2023-02-28T05:00:00.000Z'),
+      },
+      currentUpdate: {
+        path: 'service/terms-current.md',
+        content: 'content updated',
+        message: 'Record new changes of Service Terms',
+        date: new Date('2023-02-28T06:00:00.000Z'),
+      },
+    };
+
+    before(async function () {
+      this.timeout(5000);
+      git = new Git({
+        path: RECORDER_PATH,
+        author: {
+          name: config.get('recorder.versions.storage.git.author.name'),
+          email: config.get('recorder.versions.storage.git.author.email'),
+        },
+      });
+
+      await git.initialize();
+      subject = new GitRepository({
+        ...config.get('recorder.versions.storage.git'),
+        path: RECORDER_PATH,
+      });
+
+      await subject.initialize();
+
+      /* eslint-disable no-await-in-loop */
+      for (const commit of Object.values(commits)) {
+        const { path: relativeFilePath, date, content, message } = commit;
+        const filePath = path.join(RECORDER_PATH, relativeFilePath);
+
+        await GitRepository.writeFile({ filePath, content });
+
+        const sha = await subject.commit({ filePath, message, date });
```

Comment:
> I understand the only reason we now expose the `commit` method is to test it. Intrinsically, that's a bad reason, and our use of intermediary `exports.js` files to expose only subsets of features is not useful here because that method is a member of a class that is exposed or not as a whole.
>
> I would prefer finding an alternative way to test so we limit the API surface.

Reply:
I totally agree and I tried to create a fixture repository with already saved commits but I had trouble to had them in the history without having to add a sub-module.

Reply:
I have an idea, I'm going to try something else.